### PR TITLE
Add type for `Liferay`

### PR DESCRIPTION
**Pull request by jwanner83**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Review by moisout — APPROVED** [review]
Seems to follow the guidelines, as far as I can see. This would be awesome!

**Review by JannikJost — APPROVED** [review]
great work!

**Comment by jwanner83** [comment]
This type provides typescript support for Liferay's Global JavaScript Object. (See https://liferay.com / https://github.com/liferay/liferay-portal)

Although, there is a package published on the official npmjs registry with the name `liferay` (https://www.npmjs.com/package/liferay), the package is only a placeholder so that no one uses this name for a different product / tool. This means that the type doesn't conflict with the npm package because the npm package doesn't do anything but is for the same product.

**Review by JannikJost — APPROVED** [review]
👍

**Review by moisout — APPROVED** [review]

File: types/liferay/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+/// <reference path="liferay.d.ts" />
+
+export = Liferay;
```
Comment (gabritto) [inline types/liferay/index.d.ts:3]: I think, if the intent of the package is to be used as global constants, as the types indicate. you shouldn't actually export anything.
Unless users of the underlying API/source code can actually import something at run time.

Comment (gabritto) [inline types/liferay/index.d.ts:3]: I think, if the intent of the package is to be used as global constants, as the types indicate. you shouldn't actually export anything.
Unless users of the underlying API/source code can actually import something at run time.

File: types/liferay/liferay.d.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
+declare const Liferay: {
```
Comment (gabritto) [inline types/liferay/liferay.d.ts:1]: Could that be moved from this file to `index.d.ts`?

**Review by gabritto — CHANGES_REQUESTED** [review]

**Comment by gabritto** [comment]
The name conflict situation seems ok, but I left some comments about the type definitions.

File: types/liferay/index.d.ts (hunk 1 above)
Comment (jwanner83) [reply types/liferay/index.d.ts]: Thats a good point. I removed the export. Thank you for noticing!

File: types/liferay/liferay.d.ts (hunk 2 above)
Comment (jwanner83) [reply types/liferay/liferay.d.ts]: I removed the export and moved the definition to the index.d.ts. 👍

**Review by moisout — APPROVED** [review]

**Review by JannikJost — APPROVED** [review]

**Review by gabritto — APPROVED** [review]

**Comment by jwanner83** [comment]
Ready to merge

**Comment by jwanner83** [comment]
Thank you @gabritto for the review!

**Pull request merged**